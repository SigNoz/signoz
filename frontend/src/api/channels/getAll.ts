import axios from 'api';
import { ErrorResponseHandlerV2 } from 'api/ErrorResponseHandlerV2';
import { AxiosError } from 'axios';
import { ErrorV2Resp, SuccessResponseV2 } from 'types/api';
import { Channels, PayloadProps } from 'types/api/channels/getAll';

const getAll = async (): Promise<SuccessResponseV2<Channels[]>> => {
	try {
		const response = await axios.get<PayloadProps>('/channels');

		return {
			httpStatusCode: response.status,
			data: response.data.data,
		};
	} catch (error) {
		ErrorResponseHandlerV2(error as AxiosError<ErrorV2Resp>);
		throw error;
	}
};

export default getAll;
