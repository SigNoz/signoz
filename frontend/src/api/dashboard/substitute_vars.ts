import { ApiV5Instance } from 'api';
import { ErrorResponseHandlerV2 } from 'api/ErrorResponseHandlerV2';
import { QueryRangePayloadV5 } from 'api/v5/v5';
import { AxiosError } from 'axios';
import { ErrorV2Resp, SuccessResponseV2 } from 'types/api';
import { ICompositeMetricQuery } from 'types/api/alerts/compositeQuery';

interface ISubstituteVars {
	compositeQuery: ICompositeMetricQuery;
}

export const getSubstituteVars = async (
	props?: Partial<QueryRangePayloadV5>,
	signal?: AbortSignal,
	headers?: Record<string, string>,
): Promise<SuccessResponseV2<ISubstituteVars>> => {
	try {
		const response = await ApiV5Instance.post<{ data: ISubstituteVars }>(
			'/substitute_vars',
			props,
			{
				signal,
				headers,
			},
		);

		return {
			httpStatusCode: response.status,
			data: response.data.data,
		};
	} catch (error) {
		ErrorResponseHandlerV2(error as AxiosError<ErrorV2Resp>);
	}
};
