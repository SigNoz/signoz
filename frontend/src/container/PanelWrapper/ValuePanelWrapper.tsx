import GridValueComponent from 'container/GridValueComponent';
import { getUPlotChartData } from 'lib/uPlotLib/utils/getUplotChartData';

import { PanelWrapperProps } from './panelWrapper.types';

function ValuePanelWrapper({
	widget,
	queryResponse,
}: PanelWrapperProps): JSX.Element {
	const { yAxisUnit, thresholds } = widget;
	const data = getUPlotChartData(queryResponse?.data?.payload);
	const dataNew = Object.values(
		queryResponse?.data?.payload?.data?.newResult?.data?.result[0]?.table
			?.rows?.[0]?.data || {},
	);

	// this is for handling both query_range v3 and v5 responses
	const gridValueData = data?.[0].length > 0 ? data : [[0], dataNew];

	return (
		<GridValueComponent
			data={gridValueData}
			yAxisUnit={yAxisUnit}
			thresholds={thresholds}
		/>
	);
}

export default ValuePanelWrapper;
