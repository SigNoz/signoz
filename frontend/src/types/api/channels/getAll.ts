export type PayloadProps = {
	data: Channels[];
	status: string;
};

export interface Channels {
	created_at: string;
	data: string;
	id: string;
	name: string;
	type: string;
	updated_at: string;
}
