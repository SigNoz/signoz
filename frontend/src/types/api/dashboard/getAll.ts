import { PANEL_GROUP_TYPES, PANEL_TYPES } from 'constants/queryBuilder';
import { ThresholdProps } from 'container/NewWidget/RightContainer/Threshold/types';
import { timePreferenceType } from 'container/NewWidget/RightContainer/timeItems';
import { ReactNode } from 'react';
import { Layout } from 'react-grid-layout';
import { Query } from 'types/api/queryBuilder/queryBuilderData';

import { IField } from '../logs/fields';
import { BaseAutocompleteData } from '../queryBuilder/queryAutocompleteResponse';

export type PayloadProps = Dashboard[];

export const VariableQueryTypeArr = ['QUERY', 'TEXTBOX', 'CUSTOM'] as const;
export type TVariableQueryType = typeof VariableQueryTypeArr[number];

export const VariableSortTypeArr = ['DISABLED', 'ASC', 'DESC'] as const;
export type TSortVariableValuesType = typeof VariableSortTypeArr[number];

export interface IDashboardVariable {
	id: string;
	order?: any;
	name?: string; // key will be the source of truth
	description: string;
	type: TVariableQueryType;
	// Query
	queryValue?: string;
	// Custom
	customValue?: string;
	// Textbox
	textboxValue?: string;

	sort: TSortVariableValuesType;
	multiSelect: boolean;
	showALLOption: boolean;
	selectedValue?:
		| null
		| string
		| number
		| boolean
		| (string | number | boolean)[];
	// Internal use
	modificationUUID?: string;
	allSelected?: boolean;
	change?: boolean;
}
export interface Dashboard {
	id: number;
	uuid: string;
	createdAt: string;
	updatedAt: string;
	createdBy: string;
	updatedBy: string;
	data: DashboardData;
	isLocked?: boolean;
}

export interface DashboardTemplate {
	name: string;
	icon: React.ReactElement;
	id: string;
	description: string;
	previewImage: string;
}

export interface DashboardData {
	uuid?: string;
	description?: string;
	tags?: string[];
	name?: string;
	widgets?: Array<WidgetRow | Widgets>;
	title: string;
	layout?: Layout[];
	panelMap?: Record<string, { widgets: Layout[]; collapsed: boolean }>;
	variables: Record<string, IDashboardVariable>;
	version?: string;
	image?: string;
}

export interface WidgetRow {
	id: string;
	panelTypes: PANEL_GROUP_TYPES;
	title: ReactNode;
	description: string;
}

export interface ColumnUnit {
	[key: string]: string;
}
export interface IBaseWidget {
	isStacked: boolean;
	id: string;
	panelTypes: PANEL_TYPES;
	title: ReactNode;
	description: string;
	opacity: string;
	nullZeroValues: string;
	timePreferance: timePreferenceType;
	stepSize?: number;
	yAxisUnit?: string;
	stackedBarChart?: boolean;
	bucketCount?: number;
	bucketWidth?: number;
	mergeAllActiveQueries?: boolean;
	thresholds?: ThresholdProps[];
	softMin: number | null;
	softMax: number | null;
	fillSpans?: boolean;
	columnUnits?: ColumnUnit;
	selectedLogFields: IField[] | null;
	selectedTracesFields: BaseAutocompleteData[] | null;
}
export interface Widgets extends IBaseWidget {
	query: Query;
}

export interface PromQLWidgets extends IBaseWidget {
	query: { query: string; legend: string }[];
}

export interface IQueryBuilderTagFilterItems {
	id: string;
	key: string;
	op: string;
	value: string[];
}
