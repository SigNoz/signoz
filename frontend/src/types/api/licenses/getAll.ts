import { License } from './def';

export type PayloadProps = {
	licenses: License[];
};
