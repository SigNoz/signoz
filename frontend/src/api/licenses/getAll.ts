import { ApiV2Instance as axios } from 'api';
import { ErrorResponse, SuccessResponse } from 'types/api';
import { PayloadProps } from 'types/api/licenses/getAll';

const getAll = async (): Promise<
	SuccessResponse<PayloadProps> | ErrorResponse
> => {
	const response = await axios.get('/licenses');

	return {
		statusCode: 200,
		error: null,
		message: response.data.status,
		payload: response.data.data,
	};
};

export default getAll;
