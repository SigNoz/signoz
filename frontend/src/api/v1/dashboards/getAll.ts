import axios from 'api';
import { ErrorResponseHandlerV2 } from 'api/ErrorResponseHandlerV2';
import { AxiosError } from 'axios';
import { ErrorV2Resp, SuccessResponseV2 } from 'types/api';
import { Dashboard, PayloadProps } from 'types/api/dashboard/getAll';

const getAll = async (): Promise<SuccessResponseV2<Dashboard[]>> => {
	try {
		const response = await axios.get<PayloadProps>('/dashboards');
		return {
			httpStatusCode: response.status,
			data: response.data.data,
		};
	} catch (error) {
		ErrorResponseHandlerV2(error as AxiosError<ErrorV2Resp>);
	}
};

export default getAll;
