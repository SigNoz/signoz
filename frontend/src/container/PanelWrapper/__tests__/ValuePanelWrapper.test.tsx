import { render } from 'tests/test-utils';
import { Widgets } from 'types/api/dashboard/getAll';

import ValuePanelWrapper from '../ValuePanelWrapper';
import {
	thresholds,
	valuePanelQueryResponse,
	valuePanelWidget,
} from './valuePanelWrapperHelper';

describe('Value panel wrappper tests', () => {
	it('should render value panel correctly with yaxis unit', () => {
		const { getByText } = render(
			<ValuePanelWrapper
				widget={(valuePanelWidget as unknown) as Widgets}
				queryResponse={(valuePanelQueryResponse as unknown) as any}
				onDragSelect={(): void => {}}
			/>,
		);

		// selected y axis unit as miliseconds (ms)
		expect(getByText('295 ms')).toBeInTheDocument();
	});

	it('should render tooltip when there are conflicting thresholds', () => {
		const { getByTestId, container } = render(
			<ValuePanelWrapper
				widget={({ ...valuePanelWidget, thresholds } as unknown) as Widgets}
				queryResponse={(valuePanelQueryResponse as unknown) as any}
				onDragSelect={(): void => {}}
			/>,
		);

		expect(getByTestId('conflicting-thresholds')).toBeInTheDocument();
		// added snapshot test here for checking the thresholds color being applied properly
		expect(container).toMatchSnapshot();
	});
});
