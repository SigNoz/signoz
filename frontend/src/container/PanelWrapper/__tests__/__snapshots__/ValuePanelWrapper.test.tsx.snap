// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value panel wrappper tests should render tooltip when there are conflicting thresholds 1`] = `
.c1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  text-align: center;
  padding-top: 1rem;
}

<div>
   
  <div
    class="c0"
  >
    <article
      class="ant-typography css-dev-only-do-not-override-2i2tap"
    />
  </div>
  <div
    class="c1"
  >
    <div
      class="value-graph-container"
    >
      <span
        class="ant-typography value-graph-text css-dev-only-do-not-override-2i2tap"
        style="color: Blue;"
      >
        295 ms
      </span>
      <div
        class="value-graph-textconflict"
      >
        <span
          aria-label="exclamation-circle"
          class="anticon anticon-exclamation-circle value-graph-icon"
          data-testid="conflicting-thresholds"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="exclamation-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
